lib/index.ts

```diff
@@ -293,14 +297,24 @@ export async function initializeBuildMetadata(
 	secrets: SecretsPopulationMap;
 	regSecrets: RegistrySecrets;
 	architecture: string;
+	experimental: boolean;
 }> {
 	if (tasks.length === 0) {
-		return { secrets: {}, regSecrets: {}, architecture: '' };
+		return {
+			secrets: {},
+			regSecrets: {},
+			architecture: '',
+			experimental: false,
+		};
 	}
 	// This feels a bit dirty, but there doesn't seem another
 	// nicer way to do it given the current setup
 	const buildMetadata = tasks[0].buildMetadata;
-	const architecture = (await docker.version()).Arch;
+	const versionOutput = await docker.version();
```

Comment:
* The CLI already calls `docker.version()` before calling `performBuilds`, so will end up calling `docker.version()` twice. I wonder if it would make sense to take the result of `docker.version()` as an argument, instead of calling it here.
* In another comment, I question whether we should be testing for experimental. Testing the Docker API version might be more interesting. Maybe the Docker API version could be taken as argument, so there's no need to call `docker.version()` here.
* In that case, it might make more sense to get the API version as an argument to `performBuilds`, rather than an argument to `initializeBuildMetadata`, as the information is not used in this function.

Reply:
@pdcastro Great idea, let's inject the docker version and use it as a threshold to use or not experimental features. The builder does `version` when generating deltas. Is it the same case for the cli? The builder could retrieve the version following the first connection to the builder, so in that way, it knows the worker capabilities since the beginning.